pkg/pillar/types/zedagenttypes.go

```diff
@@ -264,7 +264,7 @@ const (
 	BootReasonOOM                // OOM causing process to be killed
 	BootReasonWatchdogHung       // Software watchdog due stuck agent
 	BootReasonWatchdogPid        // Software watchdog due to e.g., golang panic
-	BootReasonKernel             // TBD how we detect this
+	BootReasonKernel             // Set by dump-capture kernel, see docs/KERNEL-DUMPS.md and pkg/kdump/kdump.sh for details
```

Comment:
But code also sets it when there is no dump...

Reply:
I assume this can be always the case (reboot, no dump), which we can't avoid, since not all archs are supported and I assume crash kernel can also crash (unfortunately). So I can "relax" the comment a bit, saying ideally we rely on dump-capture kernel, but this is not always the case. Something like this.

Reply:
When it is intentionally rebooted from the controller, then we set the BootReason accordingly. But given not arches being supported I guess we either need to have this be a bit of catch all, or we introduce a new enum value for BootReasonNotLikelyKernelCrash to handle things which are unknown, but not due to a power outage.

Reply:
Will cover this in the next PR as discussed with Erik.